iml-services/iml-journal/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+#[repr(i16)]
+pub enum LogMessageClass {
+    Normal,
+    Lustre,
+    LustreError,
+    Copytool,
+    CopytoolError,
+}
+
+lazy_static! {
+    static ref LUSTRE_ERROR_TS: Regex = Regex::new(r"^\[\d+\.\d+\] LustreError:").unwrap();
+    static ref LUSTRE_TS: Regex = Regex::new(r"^\[\d+\.\d+\] Lustre:").unwrap();
+    static ref LUSTRE_ERROR: Regex = Regex::new(r"^LustreError:").unwrap();
+    static ref LUSTRE: Regex = Regex::new(r"^Lustre:").unwrap();
+}
+
+type HandlerFut<'a> = BoxFuture<'a, Result<(), ImlJournalError>>;
+type Handler = for<'a> fn(&'a PgPool, &'a str, i32, i32) -> HandlerFut<'a>;
+
+lazy_static! {
+    static ref HANDLERS: HashMap<&'static str, Handler> = {
+        let mut hm: HashMap<&'static str, Handler> = HashMap::new();
+
+        hm.insert("Can't start acceptor on port", port_used_handler);
+        hm.insert("Can't create socket:", port_used_handler);
+        hm.insert(": connection from ", client_connection_handler);
+        hm.insert(": select flavor ", server_security_flavor_handler);
+        hm.insert(
+            ": obd_export_evict_by_uuid()",
+            admin_client_eviction_handler,
+        );
+        hm.insert(": evicting client at ", client_eviction_handler);
+
+        hm
+    };
+}
+
+pub fn get_message_class(message: &str) -> LogMessageClass {
+    if LUSTRE_ERROR_TS.is_match(message) || LUSTRE_ERROR.is_match(message) {
+        LogMessageClass::LustreError
+    } else if LUSTRE_TS.is_match(message) || LUSTRE.is_match(message) {
+        LogMessageClass::Lustre
+    } else {
+        LogMessageClass::Normal
+    }
+}
+
+fn port_used_handler<'a>(
+    pool: &'a PgPool,
+    _: &str,
+    host_id: i32,
+    host_content_type_id: i32,
+) -> BoxFuture<'a, Result<(), ImlJournalError>> {
+    alert::raise(
+        pool,
+        AlertRecordType::SyslogEvent,
+        "Lustre port already being used".into(),
+        host_content_type_id,
+        None,
+        AlertSeverity::ERROR,
+        host_id,
+    )
+    .err_into()
+    .boxed()
+}
+
+fn client_connection_handler<'a>(
+    pool: &'a PgPool,
+    msg: &str,
+    host_id: i32,
+    host_content_type_id: i32,
+) -> HandlerFut<'a> {
+    if let Some((lustre_pid, msg)) = client_connection_parser(msg) {
+        return alert::raise(
+            pool,
+            AlertRecordType::ClientConnectEvent,
+            msg,
+            host_content_type_id,
+            Some(lustre_pid),
+            AlertSeverity::INFO,
+            host_id,
+        )
+        .err_into()
+        .boxed();
+    };
+
+    future::ok(()).boxed()
+}
+
+/// Parses a client connected to a target
+fn client_connection_parser(msg: &str) -> Option<(i32, String)> {
+    lazy_static! {
+        static ref TARGET_END: Regex = Regex::new(r":\s+connection from").unwrap();
+    }
+
+    // get the client NID out of the string
+    let nid_start = msg.find("@")? + 1;
+    let nid_len = msg[nid_start..].find(" ")?;
+    // and the UUID
+    let uuid_start = msg.find(" from ")? + 6;
```

Comment:
perhaps, use regex for the entire thing?

Reply:
This was pretty much a straight port from the Python equivalent (including tests). I'd rather use a parser combinator here honestly, but it's worth waiting till we have more items to match over.